elytron/src/main/java/org/wildfly/extension/elytron/FileSystemRealmDefinition.java

```diff
@@ -149,35 +228,78 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
-
             final String path = PATH.resolveModelAttribute(context, model).asString();
             final String relativeTo = RELATIVE_TO.resolveModelAttribute(context, model).asStringOrNull();
-
             final String hashEncoding = HASH_ENCODING.resolveModelAttribute(context, model).asString();
             final String hashCharset = HASH_CHARSET.resolveModelAttribute(context, model).asString();
+            final String credentialStore = CREDENTIAL_STORE.resolveModelAttribute(context, model).asStringOrNull();
+            final String secretKey = SECRET_KEY.resolveModelAttribute(context, model).asStringOrNull();
+            final String keyStoreName = KEY_STORE.resolveModelAttribute(context, model).asStringOrNull();
+            final String keyPairAlias = KEY_STORE_ALIAS.resolveModelAttribute(context, model).asStringOrNull();
 
+            final InjectedValue<KeyStore> keyStoreInjector = new InjectedValue<>();
             final InjectedValue<PathManager> pathManagerInjector = new InjectedValue<>();
             final InjectedValue<NameRewriter> nameRewriterInjector = new InjectedValue<>();
 
+            SecretKey key = null;
+            if (credentialStore != null && secretKey != null) {
+                key = getSecretKey(context, credentialStore, secretKey);
+            }
+            SecretKey finalKey = key;
+            ServiceRegistry keyStoreServiceRegistry = context.getServiceRegistry(true);
+
+
             TrivialService<SecurityRealm> fileSystemRealmService = new TrivialService<>(
                     new TrivialService.ValueSupplier<SecurityRealm>() {
-
                         private PathResolver pathResolver;
-
+                        ModifiableKeyStoreService keyStoreService;
                         @Override
                         public SecurityRealm get() throws StartException {
+
                             pathResolver = pathResolver();
                             Path rootPath = pathResolver.path(path).relativeTo(relativeTo, pathManagerInjector.getOptionalValue()).resolve().toPath();
 
                             NameRewriter nameRewriter = nameRewriterInjector.getOptionalValue();
                             Charset charset = Charset.forName(hashCharset);
                             Encoding encoding = HEX.equals(hashEncoding) ? Encoding.HEX : Encoding.BASE64;
+                            if (nameRewriter == null) {
+                                nameRewriter = NameRewriter.IDENTITY_REWRITER;
+                            }
+                            KeyStore keyStore = keyStoreInjector.getOptionalValue();
+                            PrivateKey privateKey = null;
+                            PublicKey publicKey = null;
+                            if (keyStore != null) {
+                                    try {
+                                        keyStoreService = CertificateAuthorityAccountDefinition.getModifiableKeyStoreService(keyStoreServiceRegistry, keyStoreName);
```

Comment:
It's better not to call `CertificateAuthorityAccountDefinition` here. Perhaps you could introduce a new utility class or a new utility method in an existing utility class for this.